### PR TITLE
Pu/matrixarray

### PR DESCRIPTION
**Pull request by izulin**
### Context

General naming cleanup.
Fixed docs in few places.
Fixed missing imports.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #719 
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [ ] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]
Please add changelog. We have a lot of breaking changes here:
- Error types names changed, list them
- Cell type returned by some methods has changed, list it + methods that it affects
- Config options were renamed, list them
- Public methods are renamed ie isCellPartOfArray, we should list them as breaking changes
- Does any changes may affect the custom plugin authors? ie. dependency graph, should we add internal breaking changes for them?
- AlwaysDense,  AlwaysSparse, DenseSparseChooseBasedOnThreshold, are now exported, should have Added/Fixed entry as well

Each list should have before -> after table or list. Will help users to migrate their apps

**Comment by wojciechczerniak** [comment]
Related issue #742

**Comment by izulin** [comment]
> * Does any changes may affect the custom plugin authors? ie. dependency graph, should we add internal breaking changes for them?

Should not.

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,7 +59,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added pretty print for detailedCellError. (#712)
 - Added `simpleCellRangeFromString` and `simpleCellRangeToString` helpers. (#720)
 - Added `CellError` to exports. (#736)
+<<<<<<< HEAD
```
Comment (wojciechczerniak) [inline CHANGELOG.md:62]:
```suggestion
<<<<<<< HEAD
```

🙈

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,7 +17,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - **Breaking change**: Changed behaviour of `addRows/removeRows` so that it is possible to add/remove rows across spilled array without changing array size. (#708)
 - **Breaking change**: Changed behaviour of `addColumns/removeColumns` so that it is possible to add/remove columns across spilled array without changing array size. (#732)
 - Changed SWITCH function, so it takes array as its first argument.
-- Changed TRANSPOSE function, so it works with data of any type. (#708)
+- Changed TRANSPOSE function, so it works with data of any type. (#670)
+- Changed config options (#747):
```
Comment (wojciechczerniak) [inline CHANGELOG.md:21]:
```suggestion
- **Breaking change**: Changed config options (#747):
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,7 +17,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+| before                | after                |
+|-----------------------|----------------------|
+| matrixColumnSeparator | arrayColumnSeparator |
+| matrixRowSeparator    | arrayRowSeparator    |
+
+- Changed CellType.MATRIX to CellType.ARRAY (#747)
```
Comment (wojciechczerniak) [inline CHANGELOG.md:28]:
```suggestion
- **Breaking change**: Changed CellType.MATRIX to CellType.ARRAY (#747)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,7 +17,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Changed API methods (#747):
```
Comment (wojciechczerniak) [inline CHANGELOG.md:29]:
```suggestion
- **Breaking change**: Changed API methods (#747):
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -17,7 +17,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+| before             | after             |
+|--------------------|-------------------|
+| matrixMapping      | arrrayMapping     |
+| isCellPartOfMatrix | isCellPartOfArray |
+
+- Changed Exceptions (#747):
```
Comment (wojciechczerniak) [inline CHANGELOG.md:36]:
```suggestion
- **Breaking change**: Changed Exceptions (#747):
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -17,7 +17,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - **Breaking change**: Changed behaviour of `addRows/removeRows` so that it is possible to add/remove rows across spilled array without changing array size. (#708)
 - **Breaking change**: Changed behaviour of `addColumns/removeColumns` so that it is possible to add/remove columns across spilled array without changing array size. (#732)
 - Changed SWITCH function, so it takes array as its first argument.
-- Changed TRANSPOSE function, so it works with data of any type. (#708)
+- Changed TRANSPOSE function, so it works with data of any type. (#670)
```
Comment (wojciechczerniak) [inline CHANGELOG.md:20]: Please move the braking changes before those non-breaking changes

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]
Nice! Just mark changelog items as breaking changes

File: CHANGELOG.md (hunk 6 above)
Comment (izulin) [reply CHANGELOG.md]: 👍

**Review by wojciechczerniak — APPROVED** [review]
Thanks! 🥇

File: CHANGELOG.md [hunk 7]
```diff
@@ -16,8 +16,30 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- **Breaking change**: Changed Exceptions (#747):
+
+| before                       | after                       |
+|------------------------------|-----------------------------|
+| SourceLocationHasMatrixError | SourceLocationHasArrayError |
+| TargetLocationHasMatrixError | TargetLocationHasArrayError |
+
 - Changed SWITCH function, so it takes array as its first argument.
-- Changed TRANSPOSE function, so it works with data of any type. (#708)
+- Changed TRANSPOSE function, so it works with data of any type. (#670)
```
Comment (voodoo11) [inline CHANGELOG.md:42]:
```suggestion
- Changed TRANSPOSE function, so it works with data of any type. (#708)
```

**Pull request merged**